src/Main.elm

```diff
@@ -31,8 +31,8 @@ type PageState
 
 type alias AuthedPageState =
     { user : User.User
-    , messageIds : RemoteData.WebData MessageId.Envelope
-    , threads : RemoteData.WebData Thread.Envelope
+    , messageIds : RemoteData.RemoteData Http.Error MessageId.Envelope
```

Comment:
I've noticed that you have `Data.Id.MessageId` and also `Data.MessageId` which looks a bit confusing to me. But I have no idea about gmail domain, so have no idea if it comes from there.

Reply:
The entities come with ids, so if if do `/messages` I get the pair `{id: MessageId, threadId: ThreadId}`. Doing `/threads` I get an object containing `{id: ThreadId}`. The idea was to have different types of `ID` in order to seperate an `ID` of a message from thread's `ID` but still reuse the `ID` whereas `messageId` is present and `threadId` is present.